src/metabase/public_settings.clj

```diff
@@ -205,38 +217,67 @@
       (tru "Invalid Setting: {0}/{1}" ns-symb setting-symb))
     (f)))
 
-;; TODO - it seems like it would be a nice performance win to cache this a little bit
-(defn public-settings
-  "Return a simple map of key/value pairs which represent the public settings (`MetabaseBootstrap`) for the front-end
-   application."
-  []
-  {:admin_email           (admin-email)
-   :anon_tracking_enabled (anon-tracking-enabled)
-   :available_locales     (available-locales-with-names)
-   :custom_formatting     (custom-formatting)
-   :custom_geojson        (resolve-setting 'metabase.api.geojson 'custom-geojson)
-   :email_configured      (resolve-setting 'metabase.email 'email-configured?)
-   :embedding             (enable-embedding)
-   :enable_nested_queries (enable-nested-queries)
-   :enable_query_caching  (enable-query-caching)
-   :enable_xrays          (enable-xrays)
-   :engines               (driver.u/available-drivers-info)
-   :entities              (types/types->parents :entity/*)
-   :ga_code               "UA-60817802-1"
-   :google_auth_client_id (resolve-setting 'metabase.api.session 'google-auth-client-id)
-   :has_sample_dataset    (db/exists? 'Database, :is_sample true)
-   :hide_embed_branding   (metastore/hide-embed-branding?)
-   :ldap_configured       (resolve-setting 'metabase.integrations.ldap 'ldap-configured?)
-   :map_tile_server_url   (map-tile-server-url)
-   :metastore_url         metastore/store-url
-   :password_complexity   password/active-password-complexity
-   :premium_token         (metastore/premium-embedding-token)
-   :public_sharing        (enable-public-sharing)
-   :report_timezone       (resolve-setting 'metabase.driver 'report-timezone)
-   :setup_token           (resolve-setting 'metabase.setup 'token-value)
-   :site_name             (site-name)
-   :site_url              (site-url)
-   :timezone_short        (short-timezone-name (setting/get :report-timezone))
-   :timezones             common/timezones
-   :types                 (types/types->parents :type/*)
-   :version               config/mb-version-info})
+(defsetting available-locales
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (available-locales-with-names)))
+
+(defsetting email-configured
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (resolve-setting 'metabase.email 'email-configured?)))
+
+(defsetting engines
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (driver.u/available-drivers-info)))
```

Comment:
on more note: this doesn't really matter if you're taking these out anyway but since `:getter` is a function you don't need to wrap whatever function it ultimately calls in another function. Instead of 

```clj
:getter (fn [] (driver.u/available-drivers-info))
```

you can do

```clj
:getter driver.u/available-drivers-info
```

directly